example/game/src/main/java/no/nordicsemi/android/ble/example/game/server/viewmodel/ServerViewModel.kt

```diff
@@ -0,0 +1,234 @@
+
+        var job: Job? = null
+        job = timerFinished
+            .onEach {
+                clients.value.forEach {
+                    question.correctAnswerId?.let { answer ->
+                        it.sendCorrectAnswerId(answer)
+                        _correctAnswerId.value = answer
+                    }
+                }
+                job?.cancel()
+            }
+            .launchIn(viewModelScope)
+        _selectedAnswer.value = null
+        _correctAnswerId.value = null
+        _state.value = Round(question)
+    }
+
+    private fun startServer() {
+        advertiser.startAdvertising()
+        serverManager.setServerObserver(object : ServerObserver {
+
+            override fun onServerReady() {
+                Log.w("Start Server", "Server is Ready!!")
+            }
+
+            override fun onDeviceConnectedToServer(device: BluetoothDevice) {
+                ServerConnection(context, viewModelScope, device)
+                    .apply {
+                        stateAsFlow()
+                            .onEach { connectionState ->
+                                val currentState = _state.value
+                                when (connectionState) {
+                                    ConnectionState.Ready -> {
+                                        clients.value += this
+                                        _state.value = WaitingForPlayers(clients.value.size)
+                                    }
+                                    is ConnectionState.Disconnected -> {
+                                        clients.value -= this
+
+                                        when (currentState) {
+                                            is WaitingForPlayers -> {
+                                                _state.value = WaitingForPlayers(clients.value.size)
+                                            }
+                                            else -> {
+                                                // TODO handle disconnection during game
+                                            }
+                                        }
+                                    }
+                                    else -> {}
+                                }
+                            }
+                            .launchIn(viewModelScope)
+                    }
+                    .apply {
+                        playersName
+                            .onEach { savePlayersName(it) }
+                            .launchIn(viewModelScope)
+                    }
+                    .apply {
+                        result
+                            .onEach { saveScore(it) }
+                            .launchIn(viewModelScope)
+                    }
+                    .apply {
+                        useServer(serverManager)
+
+                        val exceptionHandler = CoroutineExceptionHandler { _, throwable ->
+                            Log.e(TAG, "Error", throwable)
```

Comment:
Sometimes you do have such handlers, sometimes you don't. Is there any logic behind?